### PR TITLE
feature/IL-77/Add-OpenAPI-for-cart-package

### PR DESCRIPTION
**Pull request by shpali4**
1. Developer's full name:
Shamardin Vladislav
2. Link to Documentation:
(will be updated after merge)
3. Changes:
3.1 CartEndpoint was updated (implemented CartsApi, removed @Valid annotations from methods)
3.2 cart-openapi.yaml was added
3.3 pom.xml was updated
4. Test data 
Import cart-openapi.yaml in Postman, after authorization all responses 200 OK
5. Conclusion:
5.1 Project was built with Apache Maven successfully.
5.2 Project was tested manually with Postman successfully.

### CODE REVIEW

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java [hunk 1]
```diff
@@ -30,14 +31,14 @@
...
 @Validated
 @RequestMapping(value = CartEndpoint.CART_URL)
-public class CartEndpoint {
+public class CartEndpoint implements CartsApi {
 
     public static final String CART_URL = "/api/v1/cart";
 
     private final CartApi cartApi;
 
     @PostMapping(value = "/items")
```
Comment (vitaliibredun) [inline src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:40]: we need to  @Override the method

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java [hunk 2]
```diff
@@ -57,7 +58,7 @@ public ResponseEntity<ShoppingSessionDto> getShoppingSession(@AuthenticationPrin
     }
 
     @PatchMapping(value = "/items")
```
Comment (vitaliibredun) [inline src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:60]: we need to  @Override the method

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java [hunk 3]
```diff
@@ -69,7 +70,7 @@ public ResponseEntity<ShoppingSessionDto> updateProductsQuantityInShoppingSessio
     }
 
     @DeleteMapping(value = "/items")
```
Comment (vitaliibredun) [inline src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:72]: we need to  @Override the method

File: src/main/resources/api-specs/cart-openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,192 @@
+openapi: "3.0.3"
+
+info:
```
Comment (vitaliibredun) [inline src/main/resources/api-specs/cart-openapi.yaml:3]: we need to write in a common style as in Zufar's example in product-openapi.yaml

info:
  title: "Online-Store API"
  version: "1.0.0"

File: src/main/resources/api-specs/cart-openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        required: true
+      responses:
+        "200":
+          description: "OK"
+          content:
+            '*/*':
+              schema:
+                $ref: "#/components/schemas/ShoppingSessionDto"
+
+    delete:
```
Comment (vitaliibredun) [inline src/main/resources/api-specs/cart-openapi.yaml:57]: better to write path as in a common style

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java [hunk 6]
```diff
@@ -30,14 +31,14 @@
...
-public class CartEndpoint {
+public class CartEndpoint implements CartsApi {
 
     public static final String CART_URL = "/api/v1/cart";
 
     private final CartApi cartApi;
 
     @PostMapping(value = "/items")
-    public ResponseEntity<ShoppingSessionDto> addNewItemToShoppingSession(@RequestBody @Valid final AddNewItemsToShoppingSessionRequest request) {
+    public ResponseEntity<ShoppingSessionDto> addNewItemToShoppingSession(@RequestBody final AddNewItemsToShoppingSessionRequest request) {
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:41]: Why did you delete ‘@Valid'?
Rollback it, please

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java [hunk 7]
```diff
@@ -30,14 +31,14 @@
 @RequiredArgsConstructor
 @Validated
 @RequestMapping(value = CartEndpoint.CART_URL)
-public class CartEndpoint {
+public class CartEndpoint implements CartsApi {
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:34]: CartApi sounds better

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java [hunk 8]
```diff
@@ -57,7 +58,7 @@ public ResponseEntity<ShoppingSessionDto> getShoppingSession(@AuthenticationPrin
     }
 
     @PatchMapping(value = "/items")
-    public ResponseEntity<ShoppingSessionDto> updateProductsQuantityInShoppingSessionItem(@RequestBody @Valid final UpdateProductsQuantityInShoppingSessionItemRequest request) {
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:60]: Rollback ‘@Valid' annotation, please

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java [hunk 9]
```diff
@@ -69,7 +70,7 @@ public ResponseEntity<ShoppingSessionDto> updateProductsQuantityInShoppingSessio
     }
 
     @DeleteMapping(value = "/items")
-    public ResponseEntity<ShoppingSessionDto> deleteItemsFromShoppingSession(@RequestBody @Valid final DeleteItemsFromShoppingSessionRequest request) {
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:72]: Rollback ‘@Valid' annotation, please

File: src/main/resources/api-specs/cart-openapi.yaml [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+info:
+  title: "Online_Store API"
+  description: "Online_Store API for Cart package"
+  version: "1.0.0"
+
+servers:
+  - url: "http://localhost:8083"
+
+tags:
+  - name: "Carts"
```
Comment (Sunagatov) [inline src/main/resources/api-specs/cart-openapi.yaml:12]: Cart is better

**Review by Sunagatov — COMMENTED** [review]
Reviewed

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java (hunk 6 above)
Comment (shpali4) [reply src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:41]: With this annotation we have a trouble in case when we try to use our API not in IDE (i use Postman): Hibernate validator HV000151 "A method overriding another method must not alter the parameter constraint configuration").
In our generated interface we have the same annotation, its incorrect.
More info: https://forum.hibernate.org/viewtopic.php?p=2469903

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java (hunk 8 above)
Comment (shpali4) [reply src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:60]: same

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java (hunk 9 above)
Comment (shpali4) [reply src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:72]: same

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java (hunk 7 above)
Comment (shpali4) [reply src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:34]: We are already have CartApi in this class (import com.zufar.onlinestore.cart.api.CartApi). Is it ok to use same names?

File: src/main/resources/api-specs/cart-openapi.yaml (hunk 10 above)
Comment (shpali4) [reply src/main/resources/api-specs/cart-openapi.yaml:12]: same

File: src/main/resources/api-specs/cart-openapi.yaml (hunk 4 above)
Comment (shpali4) [reply src/main/resources/api-specs/cart-openapi.yaml:3]: fixed

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java (hunk 1 above)
Comment (shpali4) [reply src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:40]: They are overrided automaticly, isn't it?

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java (hunk 7 above)
Comment (Sunagatov) [reply src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:34]: Yes, there is no problem

File: src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java (hunk 1 above)
Comment (Sunagatov) [reply src/main/java/com/zufar/onlinestore/cart/endpoint/CartEndpoint.java:40]: using this annotation is like a best practice

**Review by Sunagatov — APPROVED** [review]
